examples/tactics/Normalization.fst

```diff
@@ -0,0 +1,50 @@
+module Normalization
+
+open FStar.Tactics
+
+let normalize (#t:Type) (x:t) : tactic unit =
+  dup;;
+  exact (quote x);;
+  norm [Delta; Primops];;
+  trefl
+
+let add_1 (x:int) : int = x + 1
+
+// add_2 is defined to be x + 1 + 1
+let add_2 (x:int) : int = synth_by_tactic (normalize (add_1 (add_1 x)))
+
+let four : int = synth_by_tactic (normalize (add_2 2))
+
+let _ = assert (four == 4)
```

Comment:
While this assert succeeds, and indeed `four` is defined exactly as `4`, node that this is checking their "semantic" equality, ie. after full normalization and SMT. Maybe we should inspect on their syntax and fail if `four` is not defined as exactly `4`. I can do that, if you'd like.